From 4162710f8750d46fb72a2d38891685ab5daeca32 Mon Sep 17 00:00:00 2001
From: Oswald Buddenhagen <oswald.buddenhagen@gmx.de>
Date: Thu, 8 Oct 2020 17:24:08 +0200
Subject: [PATCH 3/7] fixup FISH subshell setup

move binding commands out of the fish_prompt function definition, use
consistent command separators, and remove excess backslashes from the
bindings.

amends ff0fc17a.
---
 src/subshell/common.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/subshell/common.c b/src/subshell/common.c
index 1d422999e..3e1a2f7ee 100644
--- a/src/subshell/common.c
+++ b/src/subshell/common.c
@@ -1136,12 +1136,12 @@ init_subshell_precmd (char *precmd, size_t buff_size)
         break;
     case SHELL_FISH:
         g_snprintf (precmd, buff_size,
-                    " if not functions -q fish_prompt_mc;"
+                    " bind \\e" SHELL_BUFFER_KEYBINDING " 'echo (commandline)>&%d';"
+                    "bind \\e" SHELL_CURSOR_KEYBINDING " 'echo (commandline -C)>&%d';"
+                    "if not functions -q fish_prompt_mc;"
                     "functions -e fish_right_prompt;"
                     "functions -c fish_prompt fish_prompt_mc; end;"
                     "function fish_prompt;"
-                    "bind \\e\\" SHELL_BUFFER_KEYBINDING " 'echo (commandline)>&%d'\n"
-                    "bind \\e\\" SHELL_CURSOR_KEYBINDING " 'echo (commandline -C)>&%d'\n"
                     "echo \"$PWD\">&%d; fish_prompt_mc; kill -STOP %%self; end\n",
                     command_buffer_pipe[WRITE], command_buffer_pipe[WRITE], subshell_pipe[WRITE]);
         break;
-- 
2.28.0.1.g7ba5ae0684

